test/tests/functional/pbs_basil_support.py

```diff
@@ -131,69 +140,111 @@ def comp_node(self, vnode):
         try:
             pbs_node = self.server.status(NODE, id=name)[0]
         except PbsStatusError:
-            self.assertFalse(
-                "Cray compute node %s doesn't exist on pbs server" % (name))
+            self.assertFalse(pbs_node is None,
+                             "Cray compute node %s doesn't exist on pbs server"
+                             % (name))
 
         for rsc, xval in vnode.iteritems():
-            resource = 'resources_available.' + rsc
+            if rsc != 'current_aoe':
+                resource = 'resources_available.' + rsc
+            else:
+                resource = rsc
             if xval != "":
                 if resource in pbs_node:
                     rval = pbs_node[resource]
                     if rval == xval:
                         self.logger.info(
-                            "%s\t: node has %s=%s" % (name, rsc, rval))
+                            "%s: node has %s=%s" % (name, rsc, rval))
                         self.assertTrue(True)
                     else:
-                        self.assertFalse("%s\t: node has %s=%s but XML %s=%s"
+                        self.assertFalse("%s: node has %s=%s but XML %s=%s"
                                          % (name, resource, rval,
                                             rsc, xval))
                 else:
                     self.assertFalse(
                         "%s\t: node has no resource %s" % (name, rsc))
 
-    def init_inventory_node(self):
-        node = {}
-        node['vnode'] = ""
-        node['arch'] = ""
-        node['host'] = ""
-        node['mem'] = ""
-        node['ncpus'] = ""
-        node['PBScrayhost'] = ""
-        node['PBScraynid'] = ""
-        node['vntype'] = ""
-        node['accelerator_memory'] = ""
-        node['accelerator_model'] = ""
-        node['naccelerators'] = ""
-        return node
+    @classmethod
```

Comment:
Without decorator should work fine, but the codacy will report this as an issue.

Reply:
Codacy should not report this as an issue.
In fact, having this as a class method means you are making changes to the `TestBasilQuery` class instead of the python object with class `TestBasilQuery`. How PTL is handling this I don't know, but it's misleading to a future test writer looking at this code.